src/module.c

```diff
@@ -13512,6 +13540,276 @@ const char* RM_GetInternalSecret(RedisModuleCtx *ctx, size_t *len) {
     return secret;
 }
 
+
+/* --------------------------------------------------------------------------
+ * ## Config access API
+ * -------------------------------------------------------------------------- */
+
+/* Get an iterator to all configs.
+ * Optional `ctx` can be provided if use of auto-memory is desired.
+ * Optional `pattern` can be provided to filter configs by name. If `pattern` is
+ * NULL all configs will be returned.
+ *
+ * The returned iterator can be used to iterate over all configs using
+ * RedisModule_ConfigIteratorNext().
+ *
+ * Example usage:
+ * ```
+ * // Below is same as RedisModule_GetConfigIterator(ctx, NULL)
+ * RedisModuleConfigIterator *iter = RedisModule_GetConfigIterator(ctx, "*");
+ * const char *config_name = NULL;
+ * while ((config_name = RedisModule_ConfigIteratorNext(iter)) != NULL) {
+ *     RedisModuleString *value = NULL;
+ *     RedisModule_GetStringConfig(ctx, config_name, &value);
+ *     // Do something with `value`...
+ * }
+ * RedisModule_ReleaseConfigIterator(ctx, iter);
+ *
+ * // Or optionally one can use type hints to get the config value directly
+ * iter = RedisModule_GetConfigIterator(ctx, "*");
+ * RedisModuleConfigType type;
+ * while ((config_name = RedisModule_ConfigIteratorNext(iter, &type)) != NULL) {
+ *     if (type == REDISMODULE_CONFIG_TYPE_NUMERIC) {
+ *         long long value;
+ *         RedisModule_GetNumericConfig(ctx, config_name, &value);
+ *         // Do something with `value`...
+ *     }
+ * }
+ * RedisModule_ReleaseConfigIterator(ctx, iter);
+ * ```
+ *
+ * Returns a pointer to RedisModuleConfigIterator. Unless auto-memory is enabled
+ * the caller is responsible for freeing the iterator using
+ * RedisModule_ConfigIteratorFree(). */
+RedisModuleConfigIterator *RM_GetConfigIterator(RedisModuleCtx *ctx, const char *pattern) {
+    RedisModuleConfigIterator *iter = RM_Alloc(sizeof(*iter));
+    if (iter == NULL) return NULL;
+
+    iter->di = moduleGetConfigIterator();
+    if (pattern != NULL && strpbrk(pattern, "[*?"))
+        iter->pattern = sdsnew(pattern);
+    else
+         iter->pattern = NULL;
+
+    if (ctx != NULL) autoMemoryAdd(ctx,REDISMODULE_AM_CONFIG, iter);
+    return iter;
+}
+
+/* Release the iterator returned by RedisModule_GetConfigIterator(). If auto-memory
+ * is enabled and manual release is needed one must pass the same RedisModuleCtx
+ * that was used to create the iterator. */
+void RM_ReleaseConfigIterator(RedisModuleCtx *ctx, RedisModuleConfigIterator *iter) {
+    if (ctx != NULL) autoMemoryFreed(ctx,REDISMODULE_AM_CONFIG,iter);
+    moduleReleaseConfigIterator(iter->di);
+    sdsfree(iter->pattern);
+    RM_Free(iter);
```

Comment:
seems that `moduleReleaseConfigIterator()` is not needed.
```suggestion
    if (ctx != NULL) autoMemoryFreed(ctx,REDISMODULE_AM_CONFIG,iter);
    dictReleaseIterator(iter->di);
    sdsfree(iter->pattern);
    RM_Free(iter);
```